stress: failed test in cockroach/storage/storage.test: TestPushTxnHeartbeatTimeout

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/08d664015764b5b04fc09946d0588e16f8651cca

Stress build found a failed test:

```
=== RUN   TestPushTxnHeartbeatTimeout
W160413 06:25:26.094821 gossip/gossip.go:887  not connected to cluster; use --join to specify a connected node
I160413 06:25:26.094859 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160413 06:25:26.096025 storage/replica_command.go:1409  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 1s
--- FAIL: TestPushTxnHeartbeatTimeout (0.01s)
	replica_test.go:3094: 0: expSuccess=false; got pErr <nil>
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
2 runs so far, 0 failures, over 25s
8 runs so far, 0 failures, over 30s
8 runs so far, 0 failures, over 35s
8 runs so far, 0 failures, over 40s
9 runs so far, 0 failures, over 45s
9 runs so far, 0 failures, over 50s
16 runs so far, 0 failures, over 55s
16 runs so far, 0 failures, over 1m0s
16 runs so far, 0 failures, over 1m5s
17 runs so far, 0 failures, over 1m10s
17 runs so far, 0 failures, over 1m15s
24 runs so far, 0 failures, over 1m20s
24 runs so far, 0 failures, over 1m25s
24 runs so far, 0 failures, over 1m30s
25 runs so far, 0 failures, over 1m35s
25 runs so far, 0 failures, over 1m40s
32 runs so far, 0 failures, over 1m45s
32 runs so far, 0 failures, over 1m50s
32 runs so far, 0 failures, over 1m55s
33 runs so far, 0 failures, over 2m0s
33 runs so far, 0 failures, over 2m5s
39 runs so far, 0 failures, over 2m10s
39 runs so far, 0 failures, over 2m15s
39 runs so far, 0 failures, over 2m20s
40 runs so far, 0 failures, over 2m25s
40 runs so far, 0 failures, over 2m30s
46 runs so far, 0 failures, over 2m35s
46 runs so far, 0 failures, over 2m40s
46 runs so far, 0 failures, over 2m45s
47 runs so far, 0 failures, over 2m50s
47 runs so far, 0 failures, over 2m55s
53 runs so far, 0 failures, over 3m0s
53 runs so far, 0 failures, over 3m5s
54 runs so far, 0 failures, over 3m10s
54 runs so far, 0 failures, over 3m15s
54 runs so far, 0 failures, over 3m20s
59 runs so far, 0 failures, over 3m25s
59 runs so far, 0 failures, over 3m30s
59 runs so far, 0 failures, over 3m35s
60 runs so far, 0 failures, over 3m40s
60 runs so far, 0 failures, over 3m45s
64 runs so far, 0 failures, over 3m50s
64 runs so far, 0 failures, over 3m55s
64 runs so far, 0 failures, over 4m0s
65 runs so far, 0 failures, over 4m5s
65 runs so far, 0 failures, over 4m10s
68 runs so far, 0 failures, over 4m15s
69 runs so far, 0 failures, over 4m20s
70 runs so far, 0 failures, over 4m25s
70 runs so far, 0 failures, over 4m30s
70 runs so far, 0 failures, over 4m35s
72 runs so far, 0 failures, over 4m40s
73 runs so far, 0 failures, over 4m45s
74 runs so far, 0 failures, over 4m50s
75 runs so far, 0 failures, over 4m55s
75 runs so far, 0 failures, over 5m0s
75 runs so far, 0 failures, over 5m5s
78 runs so far, 0 failures, over 5m10s
78 runs so far, 0 failures, over 5m15s
79 runs completed, 1 failures, over 5m16s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -3038,7 +3038,9 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 		pushType    roachpb.PushTxnType
 		expSuccess  bool
 	}{
-		{roachpb.ZeroTimestamp, 1, roachpb.PUSH_TIMESTAMP, false}, // using 0 as time is awkward
+		// Avoid using 0 as currentTime since our manualClock is at 0 and we
+		// don't want to have outcomes depend on random logical ticks.
+		{roachpb.ZeroTimestamp, 1, roachpb.PUSH_TIMESTAMP, false},
 		{roachpb.ZeroTimestamp, 1, roachpb.PUSH_ABORT, false},
 		{roachpb.ZeroTimestamp, 1, roachpb.PUSH_TOUCH, false},
 		{roachpb.ZeroTimestamp, 1, roachpb.PUSH_QUERY, true},
@@ -3065,7 +3067,6 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 	}
 
 	for i, test := range testCases {
-		tc.manualClock.Set(1)
 		key := roachpb.Key(fmt.Sprintf("key-%d", i))
 		pushee := newTransaction(fmt.Sprintf("test-%d", i), key, 1, roachpb.SERIALIZABLE, tc.clock)
 		pusher := newTransaction("pusher", key, 1, roachpb.SERIALIZABLE, tc.clock)
@@ -3091,8 +3092,8 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 		reply, pErr := tc.SendWrapped(&args)
 
 		if test.expSuccess != (pErr == nil) {
-			t.Errorf("%d: expSuccess=%t; got pErr %s", i, test.expSuccess, pErr)
-			continue
+			t.Fatalf("%d: expSuccess=%t; got pErr %s, reply %+v", i,
+				test.expSuccess, pErr, reply)
 		}
 		if pErr != nil {
 			if _, ok := pErr.GetDetail().(*roachpb.TransactionPushError); !ok {

